### PR TITLE
Remove seed node label

### LINKED ISSUE #704: Remove seed node labels in v1.5
We have to keep seed labels present in v1.4 (seedless) for compatibility with v1.3 sidecars but we should drop those labels in v1.5.

Revert https://github.com/scylladb/scylla-operator/pull/705

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
- Remove seed node label 

**Which issue is resolved by this Pull Request:**
Resolves #704

### CODE REVIEW

**Comment by rzetelskik** [comment]
@tnozicka should we also get rid of [Replacing a dead seed node](https://operator.docs.scylladb.com/stable/nodeoperations/replace_node.html#replacing-a-dead-seed-node) section from the operator docs?

**Comment by zimnx** [comment]
> should we also get rid of [Replacing a dead seed node](https://operator.docs.scylladb.com/stable/nodeoperations/replace_node.html#replacing-a-dead-seed-node) section from the operator docs?

Yes, additional steps are no longer needed. Actually this could be a separate PR which should be backported to 1.4

File: pkg/controller/scyllacluster/resource/resource.go [hunk 1]
```diff
@@ -54,15 +54,6 @@ func MemberService(sc *scyllav1.ScyllaCluster, rackName, name string, oldService
 	labels[naming.DatacenterNameLabel] = sc.Spec.Datacenter.Name
 	labels[naming.RackNameLabel] = rackName
 
-	// TODO: Remove seed label in v1.5. (#704)
-	//       We need to keep this for v1.3 compatibility, so a ScyllaCluster that is rolling out during
-	//       the operator upgrade, has a pod that restarts or gets evicted in the meantime can become ready
-	//       as the label is still used by v1.3 sidecar before the new cluster version rolls out.
-	// If Member is seed, add the appropriate label
-	if strings.HasSuffix(name, "-0") || strings.HasSuffix(name, "-1") {
-		labels[naming.SeedLabel] = ""
```
Comment (zimnx) [inline pkg/controller/scyllacluster/resource/resource.go:63]: Please also remove definition and other usages.

Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource.go:63]: done

**Comment by rzetelskik** [comment]
Opened #727

**Review by zimnx — DISMISSED** [review]
LGTM, please rebase

**Review by zimnx — APPROVED** [review]

**Pull request merged**